### DISCUSSION TITLE
Bring back edge-clicking allowance for options in toolbar

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#26198] [Q&A] [answered]
You could easily choose options like notifications, settings etc. without precise guidance 2023.1224:

https://github.com/ppy/osu/assets/86934170/1af15759-198b-4d98-a299-2e938d589582

In today's release this feature is regressed (2023.1229):

https://github.com/ppy/osu/assets/86934170/e0b62e94-fd11-469c-b381-3a12e8ce36e7

**Comment by ghost** [comment]
I also liked the old button design more than the new one.

**Comment by tropicaaal** [comment]
Yeah, I like the new design/interactions for the header, but especially with unconventional input methods like tablet it'd be great to have the interaction area extended to the edges of the screen. (fun fact, this is apparently part of [Fitt's law](https://en.wikipedia.org/wiki/Fitts%27s_law))

**Comment by tovc** [comment]
edge clicking is the main thing that is lost in this redesign, but the space between buttons also feels like a downgrade. Windows 11's taskbar has rounded icons, but the clickable area is a square which extends to fill the entire taskbar height, and has no gaps between icons, despite there visually being space between them

rounded buttons also don't really feel like they fit well at the edge of the screen, but that's more of a subjective view

**Reply by Squiddu** [reply]
In a situation like this where rounded buttons are at the top of the screen, I'd suggest that only the bottom left/right corners be rounded with the top left/right corners staying fully square. Think of it like a bookmark-like style.

**Comment by Susko3** [comment]
My first idea for fixing this is to keep the visuals the same, but expand the click target. Works perfectly, but it looks kinda weird when the mouse is in-between two buttons:

https://github.com/ppy/osu/assets/16479013/ad748fc2-8720-49ad-9146-d201c29f1a9b

Maybe only expand the click target when it's on the edge of the screen? Red is additional clickable area, green is current (approx):

![asd](https://github.com/ppy/osu/assets/16479013/718b7be7-1ff5-464f-a1d4-de319f7cc9be)

**Reply by ghost** [reply]
What if you expand the white background that appears on hover to be the same size as the click target? Meaning you make it not rounded either.

Even better, just revert this new design to the old one.

**Reply by peppy** [reply]
The design is not being reverted. Please go ahead and PR the hit target change.

**Reply by Loreos7** [reply]
The video you've attached shows perfectly what I meant and what you should do. Just do not change what you did in the video and make it real.
Dividing green areas isn't a good idea.

**Reply by ghost** [reply]
What I meant by the way was to make it look like this:
![image](https://github.com/ppy/osu/assets/35064754/41a6e8f2-0980-4f26-90e8-8a03dacde90c)

**Reply by tropicaaal** [reply]
Honestly, I think this behavior/design is perfectly reasonable. The changed background is more to indicate selection than an actual hit target area.

**Comment by Susko3** [comment, marked as answer]
https://github.com/ppy/osu/pull/26530

**Reply by Joehuu** [reply]
Closing discussion as resolved (to remove it from `Top: All` sorting).
